mrp_unbuild_bom_cust_qty/models/mrp_unbuild.py

```diff
@@ -222,12 +222,15 @@ def _update_product_qty_from_bom_totals(self):
         ):
             return
         for unbuild in self:
-            total_product_qty = sum(
-                item.product_uom_id._compute_quantity(
-                    item.total_qty, unbuild.product_uom_id
-                )
-                for item in unbuild.bom_quants_total_ids.filtered(
-                    lambda x: x.product_uom_id.category_id == unbuild.product_uom_id.category_id
-                )
+            unbuild.product_qty = self._get_product_qty_from_bom_totals() if self._get_product_qty_from_bom_totals() > 0 else 1
```

Comment:
With this line total calculation is unnecessary made twice. Please reduce processing:
```suggestion
            total_qty = self._get_product_qty_from_bom_totals()
            unbuild.product_qty = total_qty if total_qty > 0 else 1
```